### PR TITLE
Generalise Java exception message specs

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

**Comment by eregon** [comment]
Is there an issue with the existing expectation? Is it too strict? I like that there is no random text after the execute, and that the line number is included.

**Comment by chrisseaton** [comment]
It could be `execute` or `executeAndSpecialise` depending on when how things run.

File: spec/truffle/java_exception_spec.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@
   it "includes the first lines of the Java stacktrace for uncaught Java exceptions" do
     lambda { Truffle::Debug.throw_java_exception 'message' }.should raise_error { |e|
       message = e.message.gsub(/:\d+/, ':LINE')
-      message.lines[2].should == "\torg.truffleruby.debug.TruffleDebugNodesFactory$ThrowJavaExceptionNodeFactory$ThrowJavaExceptionNodeGen.execute(TruffleDebugNodesFactory.java:LINE)\n"
+      message.lines[2].start_with?("\torg.truffleruby.debug.TruffleDebugNodesFactory$ThrowJavaExceptionNodeFactory$ThrowJavaExceptionNodeGen.execute").should be_true
```
Comment (eregon) [inline spec/truffle/java_exception_spec.rb]: How about:
```ruby
message.lines[2].should =~ /^\torg\.truffleruby\.debug\.TruffleDebugNodesFactory$ThrowJavaExceptionNodeFactory$ThrowJavaExceptionNodeGen\.\w+\(TruffleDebugNodesFactory\.java:LINE\)$/
```
Not pretty, but it's more precise and make it easier to have changing parts.
BTW, I'd like to add the start_with/end_with matchers to mspec, but not sure I'll find the time.

Comment (eregon) [inline spec/truffle/java_exception_spec.rb]: Alternatively, maybe instead add a complementary `end_with? "(TruffleDebugNodesFactory.java:LINE)\n"` expectation.

Comment (chrisseaton) [inline spec/truffle/java_exception_spec.rb]: Done.

**Pull request merged**